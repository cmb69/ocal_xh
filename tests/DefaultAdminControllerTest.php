<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Ocal_XH.
 *
 * Ocal_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Ocal_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Ocal_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Ocal;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

class DefaultAdminControllerTest extends TestCase
{
    public function testDefaultActionRendersPluginInfo(): void
    {
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $lang = $plugin_tx['ocal'];
        $systemChecker = $this->createStub(SystemChecker::class);
        $systemChecker->method('checkVersion')->willReturn(false);
        $systemChecker->method('checkExtension')->willReturn(false);
        $systemChecker->method('checkWritability')->willReturn(false);
        $sut = new DefaultAdminController("./", "", $lang, $systemChecker);
        $response = $sut->defaultAction();
        Approvals::verifyHtml($response);
    }
}
