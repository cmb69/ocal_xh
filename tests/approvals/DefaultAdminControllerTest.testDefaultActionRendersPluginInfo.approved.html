<!-- info -->
<h1>Ocal</h1>
<img src="./ocal.png" class="ocal_logo" alt="Calendar">
<p>
  Version: 1.0beta4</p>
<p>
  Copyright &copy; 2014-2017 <a href="http://3-magi.net/"
  target="_blank">Christoph M. Becker</a>
</p>
<p class="ocal_license">
  This program is free software: you can redistribute it and/or modify it
  under the terms of the GNU General Public License as published by the Free
  Software Foundation, either version 3 of the License, or (at your option)
  any later version.
</p>
<p class="ocal_license">
  This program is distributed in the hope that it will be useful, but
  <em>without any warranty</em>; without even the implied warranty of
  <em>merchantability</em> or <em>fitness for a particular purpose</em>. See
  the GNU General Public License for more details.
</p>
<p class="ocal_license">
  You should have received a copy of the GNU General Public License along with
  this program. If not, see <a href="http://www.gnu.org/licenses/"
  target="_blank">http://www.gnu.org/licenses/</a>.
</p>
<h2>System check</h2>
<ul class="ocal_syscheck">
  <li class="xh_fail">Checking that PHP version ≥ 7.1.0 … failure</li>
  <li class="xh_fail">Checking that extension 'filter' is loaded … failure</li>
  <li class="xh_fail">Checking that extension 'json' is loaded … failure</li>
  <li class="xh_fail">Checking that CMSimple_XH version ≥ 1.7.0 … failure</li>
  <li class="xh_warning">Checking that '' is writable … warning</li>
  <li class="xh_warning">Checking that './config/' is writable … warning</li>
  <li class="xh_warning">Checking that './css/' is writable … warning</li>
  <li class="xh_warning">Checking that './languages/' is writable … warning</li>
</ul>
